### ISSUE TITLE
Option to specify boolean labels?

### ISSUE THREAD

**Issue by chmac** [mozilla-services/react-jsonschema-form#80]
In some cases it would be nice to be able to provide labels instead of the strings `true` and `false`. The simplest example being `Yes` and `No` in English. I wonder if it would be practical to add this to the `uiSchema` definition? As far as I know, JSON schema doesn't provide this support.

**Comment by n1k0** [comment]
There's the `enumNames` jsonschema property which allows defining labels for enums, though for booleans that'd mean having this property without an explicit `enum` defined (as it's implicit as `[true, false]`):

```js
const schema = {type: "boolean", enumNames: ["Yes", "No"]};
const uiSchema = {"ui:widget": "radio"};
```

Odd, and unlikely to be used in existing JSON schemas out there...

Having a new `ui:labels` uiSchema property seems to be the simplest way to go:

```js
const schema = {type: "boolean"};
const uiSchema = {
  "ui:widget": "radio", 
  "ui:labels": ["Yes", "No"]
};
```

We could also add support for options to widgets:

```js
const schema = {type: "boolean"};
const uiSchema = {
  "ui:widget": {
    name: "radio", 
    labels: ["Yes", "No"]
  }
};

The latter would require a much larger internal refactoring and would be probably slightly more cumbersome to write, though the feature might be useful to other widgets as well (including custom ones).

Thoughts?
```

**Comment by n1k0** [comment]
Thinking of it a little more, I could probably live with support provided by using enumNames right from the schema. As a bonus, the patch is a three liner.

```js
const schema = {type: "boolean", enumNames: ["Yes", "No"]};
const uiSchema = {"ui:widget": "radio"};
```

**Comment by n1k0** [comment]
@chmac quickly put a patch in #81 so we can stare at code. Would this be enough to you?

**Comment by chmac** [comment]
@n1k0 Certainly looks like it would do exactly what we want. Being able to pass in the values, like Yes / No would cover most cases where we'd have thought about building our own `ui:widget`.

**Comment by n1k0** [comment]
Fixed by #81.
